libraries/libfc/test/network/test_message_buffer.cpp

```diff
@@ -79,7 +79,7 @@ BOOST_AUTO_TEST_CASE(message_buffer_growth)
     BOOST_CHECK_EQUAL(mb.bytes_to_read(), 100u);
     BOOST_CHECK_NE(mb.read_ptr(), nullptr);
     BOOST_CHECK_NE(mb.write_ptr(), nullptr);
-    BOOST_CHECK_EQUAL((mb.read_ptr() + 100), mb.write_ptr());
+    BOOST_CHECK_EQUAL(static_cast<void*>(mb.read_ptr() + 100), static_cast<void*>(mb.write_ptr()));
```

Comment:
That one is interesting, because BOOST_CHECK_EQUAL was using strncmp (and compare uninitialized memory) while we were interested in raw pointers comparison.

Reply:
I'm surprised that the test didn't fail!

Reply:
Both pointers were pointing to the same memory, so it was comparing some garbage to the same garbage :)

Reply:
Of course, makes sense, thanks @mikelik